// scripts/preprocessor.proto (V2 - 支持NER)
syntax = "proto3";

package micromodels;

// --- V1: 用于TF-IDF分类器的预处理器数据 ---
message TfidfFeatureData {
  repeated string vocabulary = 1;
  repeated float idf_weights = 2;
}

message TfidfPreprocessorData {
  TfidfFeatureData word_features = 1;
  TfidfFeatureData char_features = 2;
}


// --- V2: 用于NER序列标注模型的预处理器数据 ---
message NerPreprocessorData {
  // 使用 map 类型，key是词，value是索引。
  // Protobuf的map在序列化时并不保证顺序，但在大多数现代语言的实现中，
  // 反序列化后通常能反映原始插入顺序或按键排序。
  // 对于Rust的HashMap，顺序无关紧要，我们只关心键值对的正确性。
  map<string, int32> word_to_ix = 1;
  map<string, int32> tag_to_ix = 2;
}